### PR TITLE
Batch sarek analyses

### PR DESCRIPTION
**Pull request by b97pla**
This PR enables starting one Sarek process for an entire project rather than one process per sample (i.e. to do a `batch-analysis`).

- If passing the `--batch-analysis` argument to `ngi_pipeline_start.py` when launching the analysis for a project, all samples in the project will be started in the same nexflow process (default is to start one process per sample).

- Also, upon flow cell organization, the default `best_practice_analysis`, `pipeline` and `reference` set in Charon will be `wgs_germline`, `sarek` and `GRCh38`. These default values can be overridden on the command line.

- The sbatch stdout and stderr files will have the slurm job id in the filename instead of a timestamp.

### CODE REVIEW

File: ngi_pipeline/tests/fc_from_samplesheet.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    # iterate over the fastq files to be created and touch them
+    for fq_files in paths_from_samplesheet(samplesheet_file):
+        for fqfile in fq_files:
+            fqdir = os.path.join(unaligned, os.path.dirname(fqfile))
+            os.makedirs(fqdir, exist_ok=True)
+            open(os.path.join(fqdir, os.path.basename(fqfile)), "w").close()
+
+
+if __name__ == '__main__':
```
Comment (matrulda) [inline ngi_pipeline/tests/fc_from_samplesheet.py:41]: When is this script used? When doing manual testing?

**Review by matrulda — COMMENTED** [review]
Looks good to me! Left some minor comments.

File: ngi_pipeline/engines/sarek/models/sarek.py [hunk 2]
```diff
@@ -252,6 +252,19 @@ def _charon_status_list_from_process_status(process_status):
...
+                        sample_object=sample_object,
+                        analysis_object=analysis_object))
+            except SampleNotValidForAnalysisError as e:
+                analysis_object.log.error(e)
+
+        self.start_analysis(analysis_samples=analysis_samples, batch_analysis=batch_analysis)
+
     def analyze_sample(self, sample_object, analysis_object):
         """
         Start the analysis for the supplied NGISample object and with the analysis details contained within the supplied
```
[the remark this answers is not in the document]
Comment (b97pla) [reply ngi_pipeline/engines/sarek/models/sarek.py:270]: ah yes, that is true!

File: ngi_pipeline/engines/sarek/process.py [hunk 3]
```diff
@@ -244,8 +244,10 @@ def _slurm_script_from_command_line(
         safe_makedir(slurm_script_dir)
         slurm_script = os.path.join(
             slurm_script_dir, "{}.{}.sbatch".format(job_name, datetime.datetime.now().strftime("%s")))
-        slurm_stdout = "{}.out".format(slurm_script)
-        slurm_stderr = "{}.err".format(slurm_script)
+        slurm_stdout = os.path.join(
+            slurm_script_dir, f"{job_name}.%j.out")
```
[the remark this answers is not in the document]
Comment (b97pla) [reply ngi_pipeline/engines/sarek/process.py:248]: actually no, it is used by slurm as a placeholder for the job id. i.e. when the job starts, the job id will be inserted instead.

File: ngi_pipeline/tests/fc_from_samplesheet.py (hunk 1 above)
Comment (b97pla) [reply ngi_pipeline/tests/fc_from_samplesheet.py:41]: yes exactly, I used it to generate a file/folder structure for testing

**Comment by b97pla** [comment]
thanks for the comments! I will update that docstring

**Review by matrulda — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**